test/es6/proxybugs.js

```diff
@@ -37,6 +37,24 @@ var tests = [
                 assert.areEqual(target.x, 20, "target object should work as expected even after proxy object is added to map");
             });
         }
+    },
+    {
+        name: "Assertion validation : returning descriptor during getOwnPropertyDescriptor should not pollute the descriptor",
+        body() {
+            var target = {};
+            var handler = {};
+            handler['defineProperty'] = function () {
+                assert.fail("This function will not be called as 'getOwnPropertyDescriptor' will return a descriptor");
+            };
+            
+            handler['getOwnPropertyDescriptor'] = function (t, property) {
+                Object.defineProperty(t, 'abc', { set: function () { } });
```

Comment:
No we are not calling trap. while we are setting the value we identified there is no trap but getOwnPropertyDescriptor adds that value so we return from there (that is the fix in here).

---
In reply to: [95927783](https://github.com/Microsoft/ChakraCore/pull/2374#discussion_r95927783) [](ancestors = 95927783)